src/litDataset.test.ts

```diff
@@ -1657,3 +1659,150 @@ describe("saveLitDatasetInContainer", () => {
+        unstable_aclUrl: "https://arbitrary.pod/resource.acl",
+      },
+    };
+    const aclResource: unstable_AclDataset = Object.assign(dataset(), {
+      resourceInfo: { fetchedFrom: "https://arbitrary.pod/resource.acl" },
+      accessTo: "https://arbitrary.pod/resource",
+    });
+
+    await unstable_saveAclFor(withResourceInfo, aclResource);
+
+    expect(mockedFetcher.fetch.mock.calls).toHaveLength(1);
+  });
+
+  it("uses the given fetcher if provided", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(Promise.resolve(new Response()));
+    const withResourceInfo = {
+      resourceInfo: {
+        fetchedFrom: "https://arbitrary.pod/resource",
+        unstable_aclUrl: "https://arbitrary.pod/resource.acl",
+      },
+    };
+    const aclResource: unstable_AclDataset = Object.assign(dataset(), {
+      resourceInfo: { fetchedFrom: "https://arbitrary.pod/resource.acl" },
+      accessTo: "https://arbitrary.pod/resource",
+    });
+
+    await unstable_saveAclFor(withResourceInfo, aclResource, {
+      fetch: mockFetch,
+    });
+
+    expect(mockFetch.mock.calls).toHaveLength(1);
+  });
+
+  it("returns a meaningful error when the server returns a 403", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(
+        Promise.resolve(new Response("Not allowed", { status: 403 }))
+      );
+    const withResourceInfo = {
+      resourceInfo: {
+        fetchedFrom: "https://arbitrary.pod/resource",
+        unstable_aclUrl: "https://arbitrary.pod/resource.acl",
+      },
+    };
+    const aclResource: unstable_AclDataset = Object.assign(dataset(), {
+      resourceInfo: { fetchedFrom: "https://arbitrary.pod/resource.acl" },
+      accessTo: "https://arbitrary.pod/resource",
+    });
+
+    const fetchPromise = unstable_saveAclFor(withResourceInfo, aclResource, {
+      fetch: mockFetch,
+    });
+
+    await expect(fetchPromise).rejects.toThrow(
+      new Error("Storing the Resource failed: 403 Forbidden.")
+    );
+  });
+
+  it("marks the stored ACL as applying to the given Resource", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(Promise.resolve(new Response()));
+    const withResourceInfo = {
+      resourceInfo: {
+        fetchedFrom: "https://some.pod/resource",
+        unstable_aclUrl: "https://arbitrary.pod/resource.acl",
+      },
+    };
+    const aclResource: unstable_AclDataset = Object.assign(dataset(), {
+      resourceInfo: { fetchedFrom: "https://arbitrary.pod/resource.acl" },
+      accessTo: "https://some-other.pod/resource",
+    });
+
+    const savedAcl = await unstable_saveAclFor(withResourceInfo, aclResource, {
+      fetch: mockFetch,
+    });
+
+    expect(savedAcl.accessTo).toBe("https://some.pod/resource");
+  });
+
+  it("sends a PATCH if the ACL contains a ChangeLog and was originally fetched from the same location", async () => {
```

Comment:
I think this is a non-issue, but I'm asking just to be sure: if the ACL does not exist in the first place (its IRI as advertised by the resource Link header returns a 404), is its `fetchedFrom` property set to this IRI anyways ? And if so, a `PATCH` can create a resource, right ?

Reply:
I'm not sure if `PATCH` can create a Resource, but if it didn't exist, then this ACL will either have to have been created using `createLitDataset()` (in which case it will not have a `fetchedFrom`), or will have to have been based on a different Resource's ACL (in which case its `fetchedFrom` should be pointing to a different location), so it shouldn't be doing a `PATCH`.